### PR TITLE
Fix testing

### LINKED ISSUE #8: Fix testing
I think that during testing the app is connected to the wrong database.
- [x] Ensure all Flask extensions are initialised in `create_app`
- [x] move to `nose2`
- [x] add `nose2` layers to set up tests properly.

### PR DESCRIPTION
**Pull request by jakzale**
Closes #8.

### CODE REVIEW

File: server/app/app.py [hunk 1]
```diff
@@ -29,4 +25,13 @@ def create_app(config_name):
...
     return app
 
+def register_extensions(app):
+    """Register flask extensions"""
+    db.init_app(app)
+    bootstrap.init_app(app)
+    jwt.init_app(app)
+    migrate.init_app(app, db)
+
+    return None
```
Comment (BartKrol) [inline server/app/app.py]: I don't think you have to do return None

File: server/manage.py [hunk 2]
```diff
@@ -21,18 +20,21 @@ def make_shell_context():
...
 
-    # Remember that the first argument is always cwd
-    arguments = [os.getcwd(), server_dir]
+    # Remember that the first argument is always current file
+    arguments = [script_path, '-s', server_path]
+    plugins = ['nose2.plugins.layers']
 
     if coverage:
         arguments.append('--with-coverage')
-        arguments.append('--cover-package=app')
```
Comment (BartKrol) [inline server/manage.py:33]: What is it?

File: server/tests/layers.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+"""nose2 testing layers"""
```
Comment (BartKrol) [inline server/tests/layers.py:2]: Nice one :)

File: server/app/app.py (hunk 1 above)
Comment (jakzale) [inline server/app/app.py]: `return None` is technically unnecessary. 

Depends on the code guidelines we would like to adopt.

File: server/manage.py (hunk 2 above)
Comment (jakzale) [inline server/manage.py:33]: I have no idea

Comment (BartKrol) [inline server/manage.py:33]: Ok. We can delete it then!

**Pull request merged**